cmd/relay-gossip/main.go

```diff
@@ -99,43 +114,116 @@ var runCmd = &cli.Command{
 			}
 		}
 
-		c, err := lib.Create(cctx, cctx.IsSet(metricsFlag.Name))
+		hashesMap, err := computeHashesMap(cctx)
 		if err != nil {
-			return xerrors.Errorf("constructing client: %w", err)
+			return err
 		}
 
-		chainHash := cctx.String(lib.HashFlag.Name)
-		if chainHash == "" {
-			info, err := c.Info(context.Background())
+		for chainHash, groupConf := range hashesMap {
+			err := cctx.Set(lib.HashFlag.Name, chainHash)
 			if err != nil {
-				return xerrors.Errorf("getting chain info: %w", err)
+				return fmt.Errorf("setting client hash: %w", err)
+			}
+
+			if groupConf != "" {
+				err = cctx.Set(lib.GroupConfFlag.Name, groupConf)
+				if err != nil {
+					return fmt.Errorf("setting group-conf for hash %q got: %w", chainHash, err)
+				}
 			}
-			chainHash = hex.EncodeToString(info.Hash())
-		}
 
-		cfg := &lp2p.GossipRelayConfig{
-			ChainHash:    chainHash,
-			PeerWith:     cctx.StringSlice(peerWithFlag.Name),
-			Addr:         cctx.String(listenFlag.Name),
-			DataDir:      cctx.String(storeFlag.Name),
-			IdentityPath: cctx.String(idFlag.Name),
-			Client:       c,
+			err = boostrapGossipRelayNode(cctx, chainHash)
+			if err != nil {
+				return err
+			}
 		}
-		if _, err := lp2p.NewGossipRelayNode(log.DefaultLogger(), cfg); err != nil {
-			return err
+
+		if len(hashesMap) == 0 {
+			err := boostrapGossipRelayNode(cctx, "")
+			if err != nil {
+				return err
+			}
 		}
-		<-chan int(nil)
-		return nil
+
+		// Wait indefinitely for our client(s) to run
+		select {}
 	},
 }
 
+func boostrapGossipRelayNode(cctx *cli.Context, chainHash string) error {
+	c, err := lib.Create(cctx, cctx.IsSet(metricsFlag.Name))
+	if err != nil {
+		return fmt.Errorf("constructing client: %w", err)
+	}
+
+	chainInfo, err := c.Info(context.Background())
+	if err != nil {
+		return fmt.Errorf("cannot retrieve chain info: %w", err)
+	}
+	beaconID := chainInfo.ID
+
+	// Set the path to be desired 'storage path / beaconID'.
+	// This allows running multiple networks via the same beacon.
+	dataDir := path.Join(cctx.String(storeFlag.Name), beaconID)
+
+	cfg := &lp2p.GossipRelayConfig{
+		ChainHash:    chainHash,
+		BeaconID:     beaconID,
+		PeerWith:     cctx.StringSlice(peerWithFlag.Name),
+		Addr:         cctx.String(listenFlag.Name),
+		DataDir:      dataDir,
+		IdentityPath: cctx.String(idFlag.Name),
+		Client:       c,
+	}
+	_, err = lp2p.NewGossipRelayNode(log.DefaultLogger(), cfg)
```

Comment:
Why not annotate this error like you have been doing so you're not confused later if this does error out.